hack/quickstart-setup.sh

```diff
@@ -0,0 +1,322 @@
+    kubectl config use-context kind-${clusterName}
+    echo "Initialize local dev setup for the controller on ${clusterName}"
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}aws-credentials
+  namespace: multi-cluster-gateways
+type: "kuadrant.io/aws"
+stringData:
+  AWS_ACCESS_KEY_ID: ${MGC_AWS_ACCESS_KEY_ID}
+  AWS_SECRET_ACCESS_KEY: ${MGC_AWS_SECRET_ACCESS_KEY}
+  AWS_REGION: ${MGC_AWS_REGION}
+EOF
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}controller-config
+  namespace: multi-cluster-gateways
+data:
+  AWS_DNS_PUBLIC_ZONE_ID: ${MGC_AWS_DNS_PUBLIC_ZONE_ID}
+  ZONE_ROOT_DOMAIN: ${MGC_ZONE_ROOT_DOMAIN}
+  LOG_LEVEL: "${LOG_LEVEL}"
+EOF
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: kuadrant.io/v1alpha1
+kind: ManagedZone
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}dev-mz
+  namespace: multi-cluster-gateways
+spec:
+  id: ${MGC_AWS_DNS_PUBLIC_ZONE_ID}
+  domainName: ${MGC_ZONE_ROOT_DOMAIN}
+  description: "Dev Managed Zone"
+  dnsProviderSecretRef:
+    name: ${KIND_CLUSTER_PREFIX}aws-credentials
+    namespace: multi-cluster-gateways
+    type: AWS
+EOF
+}
+
+deployQuickStartHub() {
+    clusterName=${1}
+    kubectl config use-context kind-${clusterName}
+    echo "Initialize quickstart setup on ${clusterName}"
+    kustomize build ${QUICK_START_HUB_KUSTOMIZATION} --enable-helm --helm-command helm | kubectl apply -f -
+    echo "Waiting for ingress-nginx deployments to be ready"
+    kubectl -n ingress-nginx wait --timeout=600s --for=condition=Available deployments --all
+    echo "Waiting for metallb-system deployments to be ready"
+    kubectl -n metallb-system wait --for=condition=ready pod --selector=app=metallb --timeout=300s
+    echo "Waiting for external-dns deployments to be ready"
+    kubectl -n external-dns wait --timeout=300s --for=condition=Available deployments --all
+    echo "Waiting for cert-manager deployments to be ready"
+    kubectl -n cert-manager wait --timeout=300s --for=condition=Available deployments --all
+}
+
+deployQuickStartSpoke() {
+   clusterName=${1}
+    kubectl config use-context kind-${clusterName}
+    echo "Initialize quickstart setup on ${clusterName}"
+    kustomize build ${QUICK_START_SPOKE_KUSTOMIZATION} --enable-helm --helm-command helm | kubectl apply -f -
+    echo "Waiting for ingress-nginx deployments to be ready"
+    kubectl -n ingress-nginx wait --timeout=600s --for=condition=Available deployments --all
+    echo "Waiting for metallb-system deployments to be ready"
+    kubectl -n metallb-system wait --for=condition=ready pod --selector=app=metallb --timeout=300s
+}
+
+configureControlCluster() {
+
+    clusterName=${1}
+    # Ensure the current context points to the control plane cluster
+    kubectl config use-context kind-${clusterName}
+    kubectl label managedcluster kind-mgc-control-plane ingress-cluster=true
+    cat <<EOF | kubectl apply -f -
```

Comment:
@mikenairn I tried to include this under the config/quickstarthub kustomization but I was getting the error. `error: resource mapping not found for name: "kuadrant-multi-cluster-gateway-instance-per-cluster" namespace: "" from "STDIN": no matches for kind "GatewayClass" in version "gateway.networking.k8s.io/v1beta1"
ensure CRDs are installed first` even though I included it as the last item in the list of resources for that kustomization. And when I checked the cluster there was the usual istio gateway class present on the cluster. Any ideas on how I could fix this. It would be nice to include it in the kustomization rather than in the script like the others.

Reply:
I _think_ you are adding the gateway-api to the spoke and not the hub https://github.com/Kuadrant/multicluster-gateway-controller/pull/362/files#diff-2e5b5a1893a8b2682793d1a75d6ad60b3a3ebb132d09baee88ff5f93961eadd9R3?